diff --git a/node_modules/@react-native-community/push-notification-ios/ios/RNCPushNotificationIOS.m b/node_modules/@react-native-community/push-notification-ios/ios/RNCPushNotificationIOS.m
index 4158117..87ebc16 100644
--- a/node_modules/@react-native-community/push-notification-ios/ios/RNCPushNotificationIOS.m
+++ b/node_modules/@react-native-community/push-notification-ios/ios/RNCPushNotificationIOS.m
@@ -426,13 +426,15 @@ - (void)handleRemoteNotificationRegistrationError:(NSNotification *)notification
 }
 
 RCT_EXPORT_METHOD(onFinishRemoteNotification:(NSString *)notificationId fetchResult:(UIBackgroundFetchResult)result) {
-  RNCRemoteNotificationCallback completionHandler = self.remoteNotificationCallbacks[notificationId];
-  if (!completionHandler) {
-    RCTLogError(@"There is no completion handler with notification id: %@", notificationId);
-    return;
-  }
-  completionHandler(result);
-  [self.remoteNotificationCallbacks removeObjectForKey:notificationId];
+  dispatch_async(dispatch_get_main_queue(), ^{
+    RNCRemoteNotificationCallback completionHandler = self.remoteNotificationCallbacks[notificationId];
+    if (!completionHandler) {
+      RCTLogError(@"There is no completion handler with notification id: %@", notificationId);
+      return;
+    }
+    completionHandler(result);
+    [self.remoteNotificationCallbacks removeObjectForKey:notificationId];
+  });
 }
 
 /**
