diff --git a/node_modules/react-native-screens/android/src/main/java/com/swmansion/rnscreens/ScreenViewManager.kt b/node_modules/react-native-screens/android/src/main/java/com/swmansion/rnscreens/ScreenViewManager.kt
index 3889c2f..08784b9 100644
--- a/node_modules/react-native-screens/android/src/main/java/com/swmansion/rnscreens/ScreenViewManager.kt
+++ b/node_modules/react-native-screens/android/src/main/java/com/swmansion/rnscreens/ScreenViewManager.kt
@@ -85,7 +85,9 @@ open class ScreenViewManager :
         props: ReactStylesDiffMap?,
         stateWrapper: StateWrapper?,
     ): Any? {
-        view.setStateWrapper(stateWrapper)
+        if (stateWrapper != null) {
+            view.setStateWrapper(stateWrapper)
+        }
         return super.updateState(view, props, stateWrapper)
     }
 
