diff --git a/node_modules/react-native-reanimated/android/build.gradle b/node_modules/react-native-reanimated/android/build.gradle
index 747cc58..f89fb59 100644
--- a/node_modules/react-native-reanimated/android/build.gradle
+++ b/node_modules/react-native-reanimated/android/build.gradle
@@ -387,7 +387,8 @@ buildscript {
     }
     dependencies {
         classpath "com.android.tools.build:gradle:7.3.1"
-        classpath "de.undercouch:gradle-download-task:5.0.1"
+        // Solves the certificate issue reported at https://github.com/expo/expo/issues/20821
+        classpath "de.undercouch:gradle-download-task:4.1.2"
         classpath "com.diffplug.spotless:spotless-plugin-gradle:6.11.0"
     }
 }
diff --git a/node_modules/react-native-reanimated/android/build.gradle b/node_modules/react-native-reanimated/android/build.gradle
index 747cc58..8af333e 100644
--- a/node_modules/react-native-reanimated/android/build.gradle
+++ b/node_modules/react-native-reanimated/android/build.gradle
@@ -495,6 +495,10 @@ android {
         compileTask ->
             compileTask.dependsOn(packageNdkLibs)
     }
+    tasks.withType(com.android.build.gradle.tasks.MergeSourceSetFolders) {
+        compileTask ->
+            compileTask.dependsOn(packageNdkLibs)
+    }
     configurations {
         extractHeaders
         extractSO
diff --git a/node_modules/react-native-reanimated/ios/native/UIResponder+Reanimated.h b/node_modules/react-native-reanimated/ios/native/UIResponder+Reanimated.h
index d245c5a..9570d30 100644
--- a/node_modules/react-native-reanimated/ios/native/UIResponder+Reanimated.h
+++ b/node_modules/react-native-reanimated/ios/native/UIResponder+Reanimated.h
@@ -3,7 +3,7 @@
 #import <Foundation/Foundation.h>
 #import <React/RCTCxxBridgeDelegate.h>
 
-@interface UIResponder (Reanimated) <RCTCxxBridgeDelegate>
+@interface NSObject (Reanimated) <RCTCxxBridgeDelegate>
 
 @end
 
diff --git a/node_modules/react-native-reanimated/ios/native/UIResponder+Reanimated.mm b/node_modules/react-native-reanimated/ios/native/UIResponder+Reanimated.mm
index 4ca12f0..beeaf85 100644
--- a/node_modules/react-native-reanimated/ios/native/UIResponder+Reanimated.mm
+++ b/node_modules/react-native-reanimated/ios/native/UIResponder+Reanimated.mm
@@ -14,7 +14,7 @@
 typedef JSCExecutorFactory ExecutorFactory;
 #endif
 
-@implementation UIResponder (Reanimated)
+@implementation NSObject (Reanimated)
 - (std::unique_ptr<facebook::react::JSExecutorFactory>)jsExecutorFactoryForBridge:(RCTBridge *)bridge
 {
   const auto installer = reanimated::REAJSIExecutorRuntimeInstaller(bridge, NULL);
