diff --git a/node_modules/react-native/Libraries/Components/TextInput/TextInputState.js b/node_modules/react-native/Libraries/Components/TextInput/TextInputState.js
index 9ef598d..2e12805 100644
--- a/node_modules/react-native/Libraries/Components/TextInput/TextInputState.js
+++ b/node_modules/react-native/Libraries/Components/TextInput/TextInputState.js
@@ -32,12 +32,6 @@ function currentlyFocusedInput(): ?HostInstance {
  * If no text field is focused it returns null
  */
 function currentlyFocusedField(): ?number {
-  if (__DEV__) {
-    console.error(
-      'currentlyFocusedField is deprecated and will be removed in a future release. Use currentlyFocusedInput',
-    );
-  }
-
   return findNodeHandle<$FlowFixMe>(currentlyFocusedInputRef);
 }
 
